### ISSUE TITLE
Bug: Possible Path Resolution Issue

### ISSUE THREAD

**Issue by Saijin-Naib** [hvdwolf/jExifToolGUI#73]
I think I'm running into a PATH resolution issue on my install when I try to read/write from the Lens Configuration database.

The first time I used the Lens Configuration function, I was able to add the parameters for my Celestron C90 MAK. After I saved that Favorite Lens out, I was not able to save/load any further lenses.

SQLiteStudio can read/access the jexiftoolgui.db just fine, and it passes the Integrity Check. The Lens profile is saved.
![image](https://user-images.githubusercontent.com/19295950/94845789-299a2600-03ee-11eb-898b-d033c71e70f7.png)

My Favorite Commands can also be loaded/saved just fine, as well, which is odd, since that means that jexiftool can successfully access the database.

Userdata is going into a username with multiple spaces in the path, like "C:\Users\User Full Name\", which I think might be problematic, and the path to jexiftoolgui.exe itself is on another volume which also contains spaces, like "D:\Users\User Full Name\Desktop\Tools_Utils\exiftool\"

I could be entirely wrong here, but I'm not certain how to tell.

**Comment by hvdwolf** [comment]
There might be some issues with spaces in paths and files. Users (and I myself) already encountered them earlier. 
With user data that should not be the case as that "should" be automatically handled. That is exactly the reason why it is in the user data path.
Anyway: I will investigate.
_(And somehow my (old) Nikkor lens ended up in the "release" DB. Only the example Leica should be there. The Nikkor will be removed in next release)_

**Comment by hvdwolf** [comment]
By now I tested this with "users with spaces" on both Linux as well as Windows, and I can't reproduce it.
I also checked the sql database code, but can't find a bug in there either.
Can you please tell me if you used "special" characters? Maybe I need to escape special characters, but then I need to know what you used.

**Comment by Saijin-Naib** [comment]
I provided example paths above, stripped of my actual name, but otherwise exactly representative, ie: Spaces, but no "special" characters.

Is my JRE suspect?
![image](https://user-images.githubusercontent.com/19295950/95700747-c1c9b380-0c15-11eb-8601-a139a4cfb677.png)

**Comment by hvdwolf** [comment]
No, your JRE is OK. I normally test with 8, 11 and 14.
No issues with/between the several versions.

**Comment by Saijin-Naib** [comment]
Would you like my sqlite database to see if it is somehow corrupt?

**Comment by hvdwolf** [comment]
Can you do a `sqlite3 jexiftoolgui.db "PRAGMA integrity_check"` from a terminal or from your SqliteStudio program. That shows immediately if the the db is corrupt

**Comment by Saijin-Naib** [comment]
![image](https://user-images.githubusercontent.com/19295950/95778193-9b4b5d00-0c95-11eb-9791-5ecbf0514f96.png)
Not set Read-Only or anything, I have the required/default permissions...

And yet, I can no longer read/write from the myLenses part of the database, after storing my one record.

**Comment by hvdwolf** [comment]
OK, please let me have your database so I can check it. The first thing I will do is simply replace my own database with it to see if it works

**Comment by Saijin-Naib** [comment]
Removed my database, and jTG will allow me to load the demo lens profile just fine, even across program restarts.

With newly-created clean jTG database, after adding my lenses and re-starting, it now works. Intermittent failure of that database version somehow?

Both databases attached.  

[jexiftoolgui_data.zip](https://github.com/hvdwolf/jExifToolGUI/files/5373912/jexiftoolgui_data.zip)

**Comment by hvdwolf** [comment]
To start with:
I did a schema dump to compare: OK. No differences in database structure, so no version error either.
I did database dumps to compare, which turned out OK as well.
I did a pragma_check on both DBs: no corruption, both are fine.

However. I did find the bug in my code. When saving I ask for `lens_name` and `lens_description`, where `lens_description` is optional: that works fine. 
However, I expected `lens_description` available on retrieval and did not test for a `null` value. Your original lens in the original database did not have a `lens_description`. I now fixed it.
![lenses](https://user-images.githubusercontent.com/4520361/95981963-ce215c80-0e1f-11eb-802c-5bddc38ce832.png)

Will be in next release.
